src/plat/qemu-arm-virt/config.cmake

```diff
@@ -135,6 +135,13 @@ if(KernelPlatformQEMUArmVirt)
                 # 2 GiB, which seems a good trade-off nowadays. It's sufficient
                 # for test/demo systems, but still something the host can
                 # provide without running short on resources.
+                # The memory starts at 1 GiB (0x40000000), so up to 3 GiB can be
+                # accessed before exceeding the 32-bit address space. For 32-bit
+                # systems, using memory beyond this point is non-trivial. While
+                # the LPAE MMU model supports accessing up to a 1 TiB (40-bit)
+                # physical address space even on 32-bit systems, the 32-bit
+                # version of seL4 can access physical addresses in the 32-bit
+                # range only.
                 set(QEMU_MEMORY "1024")
```

Comment:
Is it possible to instead generate an error at build time if QEMU_MEMORY >= 3072 && KernelSel4ArchAarch32?

Reply:
That can be done when changing the scripts. But I think it would be nice to have 3072 working, so it's more intuitive what can be given there. Purpose of this PR really is not touching the scripts.

Reply:
Isn't it more likely that the new overlay does unexpected things with reserving regions outside of the 32bit region.  It would be simpler to just reject a Qemu memory configuration that the kernel doesn't support when CMake is processing.

Reply:
Actually, I think it might be a good idea to try and use the overlay method you suggest.  I can reuse the same approach for https://github.com/seL4/seL4/pull/976/files